### PR TITLE
feat(rds,secretsmanager): subnets and endpoint configuration for secret rotation

### LINKED ISSUE #17265: (rds): secret rotation application times out before rotation completes
### What is the problem?

The secret rotation application times out before rotation completes and rotation fails.

### Reproduction Steps

```ts
// create DB with generated master DB secret
const db = new rds.DatabaseInstance(this, 'Database', {
  engine: rds.DatabaseInstanceEngine.postgres({
    version: rds.PostgresEngineVersion.VER_13_4,
  }),
  instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE4_GRAVITON, ec2.InstanceSize.MICRO),
  vpc,
});
// add rotation to DB master secret
db.addRotationSingleUser({
  automaticallyAfter: cdk.Duration.days(30),
});

// create secondary DB secret (used by apps to connect to DB)
const dbUserSecret = new rds.DatabaseSecret(this, 'DbUserSecret', {
  username: 'app_user',
  masterSecret: dbAdminUserSecret,
});
// attach secondary DB secret
const dbUserSecretAttached = dbUserSecret.attach(db);
// add rotation to secondary DB secret
new secretsmanager.SecretRotation(this, 'DatabaseUserSecretRotation', {
  application: secretsmanager.SecretRotationApplication.POSTGRES_ROTATION_SINGLE_USER,
  secret: db.secret!,
  target: db,
  vpc,
  automaticallyAfter: cdk.Duration.days(30),
});
```

I log into the DB using the master user `postgres` and the generated password and create the secondary user `app_user` with password `secret_passwd`.
I'm now able to log into the DB using the user `app_user` with password `secret_passwd`.

Next, I trigger secret rotation for the secondary DB secret:

```bash
aws secretsmanager rotate-secret --secret-id <SECONDARY_DB_SECRET_ID>
```

*Alternatively:  under `AWS Console > AWS Secrets Manager > Secrets > DbUserSecretXXXXXXXX-yyyyyyyyyyyy`, press `Rotate secret immediately`*

### What did you expect to happen?

After triggering secret rotation, I can log into the DB using the newly generated password and the old password no longer works.

### What actually happened?

Secret rotation is triggered successfully, however, I'm not able to login with the newly generated password. The old password `secret_passwd` still works.

Looking at the CloudWatch Logs of the secret rotation application, I can see that the function repeatedly times out:

```
2021-11-01T17:29:04.505+01:00	START RequestId: 428afaa9-e114-43b2-8054-b71239dfb8b5 Version: $LATEST
2021-11-01T17:29:04.770+01:00	[INFO] 2021-11-01T16:29:04.769Z 428afaa9-e114-43b2-8054-b71239dfb8b5 Found credentials in environment variables.
2021-11-01T17:34:04.611+01:00	2021-11-01T16:34:04.611Z 428afaa9-e114-43b2-8054-b71239dfb8b5 Task timed out after 300.10 seconds
2021-11-01T17:34:04.611+01:00	END RequestId: 428afaa9-e114-43b2-8054-b71239dfb8b5
2021-11-01T17:34:04.611+01:00	REPORT RequestId: 428afaa9-e114-43b2-8054-b71239dfb8b5 Duration: 300100.73 ms Billed Duration: 300000 ms Memory Size: 128 MB Max Memory Used: 69 MB Init Duration: 364.56 ms
2021-11-01T17:35:45.040+01:00	START RequestId: 32583674-0f79-4c74-8971-07f594f8bb65 Version: $LATEST
2021-11-01T17:35:45.274+01:00	[INFO] 2021-11-01T16:35:45.274Z 32583674-0f79-4c74-8971-07f594f8bb65 Found credentials in environment variables.
2021-11-01T17:40:45.147+01:00	2021-11-01T16:40:45.147Z 32583674-0f79-4c74-8971-07f594f8bb65 Task timed out after 300.10 seconds
2021-11-01T17:40:45.147+01:00	END RequestId: 32583674-0f79-4c74-8971-07f594f8bb65
2021-11-01T17:40:45.147+01:00	REPORT RequestId: 32583674-0f79-4c74-8971-07f594f8bb65 Duration: 300101.37 ms Billed Duration: 300000 ms Memory Size: 128 MB Max Memory Used: 31 MB
```

Triggering rotation again gives the following error:
```
An error occurred (InvalidRequestException) when calling the RotateSecret operation: A previous rotation isn't complete. That rotation will be reattempted.
```

### CDK CLI Version

1.130.0

### Framework Version

1.130.0

### Node.js Version

v14.17.6

### OS

macOS 12.0.1

### Language

Typescript

### Language Version

4.4.4

### Other information

_No response_

### PR DESCRIPTION
**Pull request by jogold**
Add options to configure vpc subnet placement and Secrets Manager API
endpoint for the rotation Lambda function.

This is required in some VPC configurations where the database is placed
in subnets without internet connectivity.

Closes #17265

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/test/cluster.test.ts [hunk 1]
```diff
@@ -722,6 +722,227 @@ describe('cluster', () => {
...
+              'https://secretsmanager.',
+              {
+                Ref: 'AWS::Region',
+              },
+              '.',
+              {
+                Ref: 'AWS::URLSuffix',
+              },
+            ],
+          ],
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/cluster.test.ts:846]: Let's try to conserve vertical space wherever we can:

```suggestion
          'Fn::Join': ['', [
            'https://secretsmanager.',
            { Ref: 'AWS::Region' },
            '.',
            { Ref: 'AWS::URLSuffix' },
          ]],
```

Same comment in the other tests too.

File: packages/@aws-cdk/aws-rds/test/instance.test.ts [hunk 2]
```diff
@@ -7,9 +7,9 @@ import * as kms from '@aws-cdk/aws-kms';
 import * as lambda from '@aws-cdk/aws-lambda';
 import * as logs from '@aws-cdk/aws-logs';
 import * as s3 from '@aws-cdk/aws-s3';
+import { testFutureBehavior } from '@aws-cdk/cdk-build-tools/lib/feature-flag';
 import * as cdk from '@aws-cdk/core';
 import * as cxapi from '@aws-cdk/cx-api';
-import { testFutureBehavior } from '@aws-cdk/cdk-build-tools/lib/feature-flag';
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/instance.test.ts:12]: Same comment here (let's revert this change).

File: packages/@aws-cdk/aws-rds/test/instance.test.ts [hunk 3]
```diff
@@ -738,6 +738,214 @@ describe('instance', () => {
...
+    expect(stack).toHaveResourceLike('AWS::Serverless::Application', {
+      Parameters: {
+        masterSecretArn: {
+          Ref: 'DatabaseSecretAttachmentE5D1B020',
+        },
+      },
+    });
+  });
+
+  test('addRotationSingleUser() with options', () => {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/instance.test.ts:804]:
```suggestion
  test('addRotationSingleUser() with VPC Subnets', () => {
```

File: packages/@aws-cdk/aws-rds/test/cluster.test.ts [hunk 4]
```diff
@@ -722,6 +722,227 @@ describe('cluster', () => {
...
+    expect(stack).toHaveResourceLike('AWS::Serverless::Application', {
+      Parameters: {
+        masterSecretArn: {
+          Ref: 'DatabaseSecretAttachmentE5D1B020',
+        },
+      },
+    });
+  });
+
+  test('addRotationSingleUser() with options', () => {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/cluster.test.ts:795]:
```suggestion
  test('addRotationSingleUser() with VPC Subnets', () => {
```

**Review by skinny85 — CHANGES_REQUESTED** [review]
Looks great as always @jogold. Minor nitpicky comments about not very important things (also as usual, but this time from me 😜).

File: packages/@aws-cdk/aws-rds/test/instance.test.ts (hunk 3 above)
Comment (jogold) [reply packages/@aws-cdk/aws-rds/test/instance.test.ts:804]: The other options (`automaticallyAfter` and `excludeCharacters`) are tested here too.

File: packages/@aws-cdk/aws-rds/test/cluster.test.ts (hunk 4 above)
Comment (jogold) [reply packages/@aws-cdk/aws-rds/test/cluster.test.ts:795]: The other options (`automaticallyAfter` and `excludeCharacters`) are tested here too.

**Review by skinny85 — APPROVED** [review]
